src/async-wrap.cc

```diff
@@ -201,6 +332,7 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
   SET_HOOK_FN(before);
   SET_HOOK_FN(after);
   SET_HOOK_FN(destroy);
+  env->isolate()->SetPromiseHook(PromiseHook);
```

Comment:
If I'm reading this correct, this means that we only have domain support for promises when `async_wrap` is enabled. That might be okay if we enable `async_wrap` when `domain` is used, but as it is now this makes the behavior of `domain` depend on the state of `async_hooks`.